common/containers/Tabs/Swap/components/CurrencySwap.js

```diff
@@ -1,64 +1,44 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import translate from 'translations';
-import { combineAndUpper } from 'api/bity';
-
-class CoinTypeDropDown extends Component {
-  constructor(props) {
-    super(props);
-  }
-
-  static propTypes = {
-    kind: PropTypes.any,
-    onChange: PropTypes.any,
-    kindOptions: PropTypes.any
-  };
-
-  render() {
-    return (
-      <span className="dropdown">
-        <select
-          value={this.props.kind}
-          className="btn btn-default"
-          onChange={this.props.onChange.bind(this)}
-        >
-          {this.props.kindOptions.map((obj, i) => {
-            return <option value={obj} key={i}>{obj}</option>;
-          })}
-        </select>
-      </span>
-    );
-  }
-}
+import { combineAndUpper } from 'utils/formatters';
+import * as swapTypes from 'actions/swapTypes';
+import SimpleDropDown from 'components/ui/SimpleDropdown';
+
+export type CurrencySwapReduxStateProps = {
+  bityRates: {},
+  originAmount: string | number,
+  destinationAmount: string | number,
+  originKind: string,
+  destinationKind: string,
+  destinationKindOptions: String[],
+  originKindOptions: String[]
+};
+
+export type CurrencySwapReduxActionProps = {
+  originKindSwap: (value: string) => swapTypes.OriginKindSwapAction,
+  destinationKindSwap: (value: string) => swapTypes.DestinationKindSwapAction,
+  originAmountSwap: (
+    value: string | number
+  ) => swapTypes.OriginAmountSwapAction,
+  destinationAmountSwap: (
+    value: string | number
+  ) => swapTypes.DestinationAmountSwapAction,
+  changeStepSwap: () => swapTypes.ChangeStepSwapAction
+};
 
 export default class CurrencySwap extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      disabled: false
-    };
-  }
+  props: CurrencySwapReduxStateProps & CurrencySwapReduxActionProps;
 
-  static propTypes = {
-    bityRates: PropTypes.any,
-    originAmount: PropTypes.any,
-    destinationAmount: PropTypes.any,
-    originKind: PropTypes.string,
-    destinationKind: PropTypes.string,
-    destinationKindOptions: PropTypes.array,
-    originKindOptions: PropTypes.array,
-    originKindSwap: PropTypes.func,
-    destinationKindSwap: PropTypes.func,
-    originAmountSwap: PropTypes.func,
-    destinationAmountSwap: PropTypes.func,
-    partOneCompleteSwap: PropTypes.func
+  state = {
+    disabled: false
   };
 
   onClickStartSwap = () => {
-    this.props.partOneCompleteSwap(true);
+    this.props.changeStepSwap(2);
```

Comment:
Gonna leave for now. `changeStepSwap(2)` is pretty readable imo (unless you think of something specific thats better)